docs/samples/deployments/spark/presidio_anonymize_blobs.py

```diff
@@ -0,0 +1,153 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Anonymize PII Entities in text files
+# MAGIC
+# MAGIC <br>Using Presidio, anonymize PII content of files in an Azure Storage account.
+# MAGIC
+# MAGIC <br>The following code sample will:
+# MAGIC <ol>
+# MAGIC <li>Import the content of text files located in an Azure Storage blob folder</li> # noqa D501
+# MAGIC <li>Anonymize the content using Presidio</li>
+# MAGIC <li>Write the anonymized content back to the Azure Storage blob account</li>
+# MAGIC </ol>
+
+# COMMAND ----------
+
+from azure.storage.blob import BlobServiceClient
+from presidio_analyzer import AnalyzerEngine
+from presidio_anonymizer import AnonymizerEngine
+from presidio_anonymizer.entities import AnonymizerConfig
+from pyspark.sql.types import StringType
+from pyspark.sql.functions import input_file_name, regexp_replace
+from pyspark.sql.functions import col, pandas_udf
+import pandas as pd
+
+dbutils.widgets.text(
+    "storage_account_name", "", "Blob Storage Account Name"
+)  # noqa: F821
+dbutils.widgets.text("storage_container_name", "", "Blob Container Name")  # noqa: F821
+dbutils.widgets.text(
+    "storage_account_access_key", "", "Storage Account Access Key"
+)  # noqa: F821
+dbutils.widgets.text(
+    "storage_input_folder", "input", "Storage Account Access Key"
+)  # noqa: F821
+dbutils.widgets.text(
+    "storage_output_folder", "output", "Storage Account Access Key"
+)  # noqa: F821
+
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # Import the text files from Azure Blob storage
+
+# COMMAND ----------
+
+storage_account_name = dbutils.widgets.get("storage_account_name")  # noqa F821
+storage_container_name = dbutils.widgets.get("storage_container_name")  # noqa F821
+storage_account_access_key = dbutils.widgets.get(
+    "storage_account_access_key"
+)  # noqa F821
+
+
+blob_service_client = BlobServiceClient(
+    account_url="https://" + storage_account_name + ".blob.core.windows.net/",
+    credential=storage_account_access_key,
+)
+container_client = blob_service_client.get_container_client(storage_container_name)
+
+blob_names = container_client.list_blobs(
+    name_starts_with=dbutils.widgets.get("storage_input_folder") + "/"
+)
+blobs = []
+for blob in blob_names:
+    blobs.append(
+        "wasbs://"
+        + storage_container_name
+        + "@"
+        + storage_account_name
+        + ".blob.core.windows.net/"
+        + blob.name
+    )
+
+spark.conf.set(
+    "fs.azure.account.key." + storage_account_name + ".blob.core.windows.net",
+    storage_account_access_key,
+)
+
+input_df = spark.read.text(blobs).withColumn("filename", input_file_name())
+display(input_df)
+
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC # Anonymize Text using Presidio
+
+# COMMAND ----------
+
+
+def anonymize_text(text: str) -> str:
+    analyzer = AnalyzerEngine()
```

Comment:
Since @omri374 mentioned the `AnalyzerEngine` is heavy, should we instantiate it outside `anonymize_text()` and [broadcast](http://people.duke.edu/~ccc14/sta-663-2018/notebooks/S15E_Efficiency_In_Spark.html#Broadcast-Variables) it?

Reply:
can't load spacy models in parallel due to a limitation. commented in code

Reply:
Noticed that the `presidio-analyzer` package [uses spaCy 2.3](https://github.com/microsoft/presidio/blob/bf46a61a6d06ad47f3c12b5cd975423ff767f243/presidio-analyzer/Pipfile#L7) Is it possible to update `presidio-analyzer` to use [spaCy v2.3.5](https://github.com/explosion/spaCy/releases/tag/v2.3.5) or higher? The reason I think its worth trying is because the parallel thread issue might have been fixed in the thinc dependency as [stated here](`presidio-analyzer`) and spaCy v2.3.5 contains a higher version of thinc.

Reply:
@omri374  worth a check?

Reply:
we can defintly add it to our backlog

Reply:
@navalev @balteravishay worth looking into, but instead of pushing a new version with spacy 2.3.5 we can first check by just pip installing spacy 2.3.5 on Databricks and see if it solves the issue. @balteravishay WDYT? worth the try? I don't want to keep this PR from closing so could be a future suggestion.